[Tooling] Update `release-toolkit` to fix milestone moving issue

iangmaia opened the pull request:
Follow-up to AINFRA-1359

### Description

This PR uses the [13.5.1](https://github.com/wordpress-mobile/release-toolkit/releases/tag/13.5.1) release-toolkit version that fixes the action `update_assigned_milestone` to work with fine-grained GitHub tokens (see PR https://github.com/wordpress-mobile/release-toolkit/pull/663).

iangmaia committed:
Update release-toolkit to fix milestone moving problem

AliSoftware commented on Gemfile:
```diff
@@ -16,10 +16,10 @@ gem 'fastlane-plugin-sentry', '~> 1.0'
-# Using '~> 13.3.1' would constrain us to 13.3.2, 13.3.3, etc. without ever going up to 13.4
-# We need 13.3.1 because of a fix in screenshots generation we depend upon.
-gem 'fastlane-plugin-wpmreleasetoolkit', '~> 13.3', '>= 13.3.1'
+# The '>= 13.5.1' after '~> 13.5' ensures that we resolve to any version compatible with 13.5 starting from 13.5.1
+# Using '~> 13.5.1' would constrain us to 13.5.2, 13.5.3, etc. without ever going up to 13.6
+# We need 13.5.1 because of a fix in fetching issues / PRs with a fine-grained GitHub token
+gem 'fastlane-plugin-wpmreleasetoolkit', '~> 13.5', '>= 13.5.1'
```
Same non-blocker, FYI note as in https://github.com/woocommerce/woocommerce-android/pull/14715#discussion_r2412106440 🙂

iangmaia committed:
Remove `>= 13.5.1` requirement and clean Gemfile and run `bundle update fastlane-plugin-wpmreleasetoolkit`

h/t @AliSoftware https://github.com/woocommerce/woocommerce-android/pull/14715#discussion_r2412106440
